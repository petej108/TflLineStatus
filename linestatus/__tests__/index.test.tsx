import { render, screen } from '@testing-library/react';
import Home from '@/pages/index';
import lineInfoApiDataMock from '../data/lineStatusMock.json';

jest.mock('../components/useLineStatusRequest', () => ({
  useGetLineStatus: () => ({
    data: lineInfoApiDataMock,
    error: null,
  }),
}));

describe('Line Status Home', () => {
  it('renders a heading', () => {
    render(<Home />);

    const heading = screen.getByRole('heading', {
      name: /Line Status/,
    });

    expect(heading).toBeInTheDocument();
  });

  
});
