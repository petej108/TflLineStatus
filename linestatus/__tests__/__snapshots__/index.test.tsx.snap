// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line Status Home renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Line Status
      </h1>
      <p
        class="description"
      >
        Current status of London underground lines.
      </p>
      <button>
        Refresh Data
      </button>
      <ul
        class="grid gridOuter"
      >
        <li
          class="card grid"
          id="bakerloo"
          tabindex="0"
        >
          <div>
            Bakerloo
          </div>
           
          <div>
            Part Closure
          </div>
        </li>
        <li
          class="card grid"
          id="central"
          tabindex="0"
        >
          <div>
            Central
          </div>
           
          <div>
            Good Service
          </div>
        </li>
        <li
          class="card grid"
          id="circle"
          tabindex="0"
        >
          <div>
            Circle
          </div>
           
          <div>
            Severe Delays
          </div>
        </li>
        <li
          class="card grid"
          id="district"
          tabindex="0"
        >
          <div>
            District
          </div>
           
          <div>
            Good Service
          </div>
        </li>
        <li
          class="card grid"
          id="hammersmith-city"
          tabindex="0"
        >
          <div>
            Hammersmith & City
          </div>
           
          <div>
            Severe Delays
          </div>
        </li>
        <li
          class="card grid"
          id="jubilee"
          tabindex="0"
        >
          <div>
            Jubilee
          </div>
           
          <div>
            Good Service
          </div>
        </li>
        <li
          class="card grid"
          id="metropolitan"
          tabindex="0"
        >
          <div>
            Metropolitan
          </div>
           
          <div>
            Good Service
          </div>
        </li>
        <li
          class="card grid"
          id="northern"
          tabindex="0"
        >
          <div>
            Northern
          </div>
           
          <div>
            Good Service
          </div>
        </li>
        <li
          class="card grid"
          id="piccadilly"
          tabindex="0"
        >
          <div>
            Piccadilly
          </div>
           
          <div>
            Severe Delays
          </div>
        </li>
        <li
          class="card grid"
          id="victoria"
          tabindex="0"
        >
          <div>
            Victoria
          </div>
           
          <div>
            Good Service
          </div>
        </li>
        <li
          class="card grid"
          id="waterloo-city"
          tabindex="0"
        >
          <div>
            Waterloo & City
          </div>
           
          <div>
            Planned Closure
          </div>
        </li>
      </ul>
    </main>
    <footer
      class="footer"
    />
  </div>
</div>
`;
